extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -201,30 +209,31 @@ class LifecycleTask extends BackbeatTask {
             params.Marker = bucketData.details.marker;
         }
 
-        const req = this.s3target.listObjects(params);
-        attachReqUids(req, log);
+        const command = new ListObjectsCommand(params);
+        attachReqUids(command, log);
         async.waterfall([
-            next => req.send((err, data) => {
-                LifecycleMetrics.onS3Request(log, 'listObjects', 'bucket', err);
-
-                if (err) {
+            next => this.s3target.send(command)
+                .then(data => {
+                    LifecycleMetrics.onS3Request(log, 'listObjects', 'bucket', null);
+                    return next(null, data);
+                })
+                .catch(err => {
+                    LifecycleMetrics.onS3Request(log, 'listObjects', 'bucket', err);
                     log.error('error listing bucket objects', {
                         method: 'LifecycleTask._getObjectList',
                         error: err,
                         bucket: params.Bucket,
                     });
                     return next(err);
-                }
-
-                return next(null, data);
-            }),
+                }),
             (data, next) => {
                 if (data.IsTruncated && nbRetries === 0) {
                     // re-queue to Kafka topic bucketTasksTopic
                     // with bucket name and `data.marker` only once.
                     let marker = data.NextMarker;
-                    if (!marker && data.Contents.length > 0) {
-                        marker = data.Contents[data.Contents.length - 1].Key;
+                    const contents = data.Contents || [];
```

Comment:
move `contents` var up, so it can be used in the call to _compareRulesToList() at the end of the function
```suggestion
const contents = data.Contents || [];
if (data.IsTruncated && nbRetries === 0) {
	// re-queue to Kafka topic bucketTasksTopic
	// with bucket name and `data.marker` only once.
	let marker = data.NextMarker;
```